src/SourceCode.Clay/SecureStringExtensions.cs

```diff
@@ -38,10 +38,23 @@ public static string ToUnsecureString(this SecureString ss)
             }
             finally
             {
-                Marshal.ZeroFreeGlobalAllocUnicode(ptr);
+                if (ptr != IntPtr.Zero)
+                    Marshal.ZeroFreeGlobalAllocUnicode(ptr);
```

Comment:
Zero the memory first.

Reply:
`ZeroFreeGlobalAllocUnicode` already does that